lib/rspec/core/memoized_helpers.rb

```diff
@@ -144,6 +146,25 @@ def __init_memoized
                       end
       end
 
+      # @private
+      def enforce_value_expectation(matcher, method_name)
+        matcher_supports_value_expectations = begin
+                                                matcher.supports_value_expectations?
+                                              rescue
+                                                true
+                                              end
+        return if matcher_supports_value_expectations
```

Comment:
This just looks ugly 😂 , WDYT about extracting the block to a method e.g:

```suggestion
        return if matcher_supports_value_expectations?
```

and elsewhere:

```
      # @private
      def  matcher_supports_value_expectations?
        matcher.supports_value_expectations?
      rescue
        true
      end
```

Reply:
No objections. Especially since `enforce_value_expectation` will remain only in RSpec 3.x, and will be removed along with `should` in 4.
Fixed.